lib/activerecord_sqlserver_adapter/jdbc/dataset.rb

```diff
@@ -0,0 +1,225 @@
+module ActiveRecordSqlServerAdapter
+  module Jdbc
+    class Dataset
+      include Enumerable
+
+      OPTS = {}.freeze
+
+      def synchronize(&block)
+        @cache_mutex.synchronize(&block)
+      end
```

Comment:
> but our ruby jdbc wrapper is not.
> Maybe this is not necessary because AR does pooling anyways, but I'm not sure.

Exactly! We don't have to do this for them. It is code Sequl has to do, but again, not us.